sdk/nodejs/resource.ts

```diff
@@ -269,6 +269,13 @@ export abstract class CustomResource extends Resource {
     // tslint:disable-next-line:variable-name
     public readonly __pulumiCustomResource: boolean;
 
+    /**
+     * Private field containing the type ID for this object. Useful for implementing `isInstance` on
+     * classes that inherit from `CustomResource`.
+     */
+    // tslint:disable-next-line:variable-name
+    /* @internal */ public readonly __pulumiType: string;
```

Comment:
This is interesting in its own right - do we need to make tfgen/kubernetes changes to accommodate this?

Reply:
Yes we do.

Reply:
👍